### ISSUE TITLE
Problem with resource copying on linux

### ISSUE THREAD

**Issue by v1993** [cocos2d/cocos2d-x#18813]
- cocos2d-x version:git
- devices test on: linux mint 18.3
- developing environments
   - NDK version:
   - Xcode version:
   - VS version:
   - browser type and version:

Steps to Reproduce:

1. Build game on linux.
2. Change  some resource files.
3. Build again, and resources will NOT be updated.

What could be a problem?

**Comment by v1993** [comment]
Related code:

    function()    function(cocos_copy_target_res cocos_target)
        set(oneValueArgs COPY_TO)
        set(multiValueArgs FILES FOLDERS)
        cmake_parse_arguments(opt "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
        # copy files
        foreach(cc_file ${opt_FILES})
            get_filename_component(file_name ${cc_file} NAME)
            add_custom_command(TARGET ${cocos_target} PRE_BUILD
                               COMMAND ${CMAKE_COMMAND} -E copy_if_different
                               ${cc_file}
                               "${opt_COPY_TO}/${file_name}"
                               )
        endforeach()
        # copy folders
        foreach(cc_folder ${opt_FOLDERS})
            add_custom_command(TARGET ${cocos_target} PRE_BUILD
                               COMMAND ${CMAKE_COMMAND} -E copy_directory
                               ${cc_folder}
                               ${opt_COPY_TO}
                               )
        endforeach()

    endfunction()

**Comment by drelaptop** [comment]
thanks for report this, I don't know why CMake don't copy folder again. the PRE_BUILD custom command looks good. Anyway I will research this.

**Comment by v1993** [comment]
I've checked it on cocos tests, also occurs. Replacing `copy_if_different` with `copy` didn't helped.

**Comment by drelaptop** [comment]
I think we can use `configure_file ` instead. refer to

https://stackoverflow.com/questions/34799916/cmake-copy-file-from-source-directory-to-binary-directory

**Comment by drelaptop** [comment]
@v1993 Fixed in #18816 , tested using 2 different way, it works well. Could you test it again?

One of my tests:
```sh
cd cocos2d-x
mkdir linux-build && cd linux-build
cmake ..
build cpp-tests
# run cpp-tests
# modify tests/cpp-tests/Resources/strings/LabelFNTUNICODELanguages.xml
build cpp-tests # again
# run cpp-tests, see the difference caused by LabelFNTUNICODELanguages.xml
# Label changed, Label not changed when using the older cmake build
```
